api/search.js

```diff
@@ -31,120 +31,123 @@ function setup( addressDbPath, streetDbPath ){
     if( 'string' !== typeof number ){ return cb( 'invalid number' ); }
     if( 'string' !== typeof street ){ return cb( 'invalid street' ); }
 
-    var normalized = {
-      number: analyze.housenumber( number ),
-      street: analyze.street( street )
-    };
+    analyze.street(street, function streetAnalyzeCallback(err, street, metadata) {
```

Comment:
What's the idea behind this syntax? It seems... verbose

```javascript
analyze.street(street, function streetAnalyzeCallback(err, street, metadata) {
```

wouldn't that be simpler as:

```javascript
analyze.street(street, (err, street, metadata) => {
```

I see that `streetAnalyzeCallback` isn't referenced anywhere else in the file so I'm guessing it's for stack traces?

Reply:
I see in another file below that you're using the syntax:

```javascript
const analyze_street = util.promisify(analyze.street);
var names = await analyze_street();
```

I think that syntax is much cleaner, and it allows you to leave most of the code untouched and avoids the closure?

Reply:
This was simply because I was hoping to avoid having to use `async/await` syntax at all in this PR. Despite the verboseness I'm honestly not a fan of additional language complexity. Functions are well understood by everyone :)

However, because later on we are depending on many calls to libpostal returning before proceeding, using `Promisify.all` is basically the only way to get the behavior we want.

Since we're already going to be using the new syntax, I'll update this code to use it as well.